src/libusermode/userhook.cpp

```diff
@@ -187,6 +188,36 @@ static dll_t* get_pending_dll(drakvuf_t drakvuf, drakvuf_trap_info* info, userho
     return nullptr;
 }
 
+int inject_copy_memory(userhook* plugin, drakvuf_t drakvuf,
+    drakvuf_trap_info_t* info, addr_t addr)
+{
+    uint64_t buffer = 0xdeadbeafdeadbeaf;
+    uint64_t read_bytes = 0xc0febabec0febabe;
+    struct argument args[7] = {};
+    init_int_argument(&args[0], info->attached_proc_data.base_addr);
+    init_int_argument(&args[1], addr);
+    init_int_argument(&args[2], info->attached_proc_data.base_addr);
+    init_struct_argument(&args[3], buffer);
+    init_int_argument(&args[4], sizeof(buffer));
+    init_int_argument(&args[5], 0); // UserMode (TODO Is this correct?)
```

Comment:
looking through various places, setting this to UserMode (which appears to be `1`, not `0`) would only add additional checks with [ProbeForRead](https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/wdm/nf-wdm-probeforread) and [ProbeForWrite](https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/wdm/nf-wdm-probeforwrite). I believe we should keep this at KernelMode (so as is `0`). Only thing would document this with code comment.

Reply:
Thanks. That is old comment. I remove it.